test/onnx/test_op_consistency.py

```diff
@@ -0,0 +1,462 @@
+                    sample_num=i,
+                    input=repr(cpu_sample.input),
+                    args=repr(cpu_sample.args),
+                    kwargs=repr(cpu_sample.kwargs),
+                ):
+                    model = SingleOpModel(op, cpu_sample.kwargs)
+                    model.eval()
+
+                    # Run the test
+                    inputs = (cpu_sample.input, *cpu_sample.args)
+
+                    if dtype == torch.bfloat16:
+                        # Only export to ONNX without running with onnxruntime because
+                        # the CPU execution path for bfloat16 is not implemented in onnxruntime.
+                        model_buffer = io.BytesIO()
+                        torch.onnx.export(
+                            model, inputs, model_buffer, opset_version=opset
+                        )
+                        model_buffer.seek(0)
+                        onnx_model = onnx.load(model_buffer)
+                        onnx.checker.check_model(onnx_model, full_check=True)
+                        continue
+
+                    verification.verify(
+                        model,
+                        inputs,
+                        input_kwargs={},
+                        opset_version=opset,
+                        keep_initializers_as_inputs=True,
+                        ort_providers=ORT_PROVIDERS,
+                        check_shape=True,
+                        check_dtype=True,
+                        flatten=True,
+                    )
+
+        return _output_match_base
+
+    @classmethod
+    def parameterize_opsets(cls, opsets: Sequence[int]):
+        """Parameterizes the TestConsistency class with the given opsets."""
+        for opset in opsets:
+            # Generate a test method for each opset
+            test_name = f"test_output_match_opset_{opset}"
+            base_method = cls.create_test_base(opset)
+            # Important to rename the test method so that DecorateInfo can find it
+            base_method.__name__ = test_name
+
+            # Update the ops to skip in the OpInfo database
+            decorated = skip_ops(
+                OPS_DB,
+                cls.__name__,
+                test_name,
+                opset=opset,
+                skip_or_xfails=EXPECTED_SKIPS_OR_FAILS,
+            )(base_method)
+
+            # Create parameterized tests for each op
+            if opset < 13:
+                # bfloat16 is not supported before opset 13
+                allowed_dtypes = tuple(
+                    [dtype for dtype in SUPPORTED_DTYPES if dtype != torch.bfloat16]
+                )
+            else:
+                allowed_dtypes = SUPPORTED_DTYPES
+            filtered_ops = [op for op in OPS_DB if op.name in ALLOWLIST_OP]
+            decorated = common_device_type.ops(
+                filtered_ops,
+                allowed_dtypes=allowed_dtypes,
+            )(decorated)
+
+            setattr(cls, test_name, decorated)
```

Comment:
I am confused about what is going on here. `decorated` is assigned, then passed to `common_device_type.ops` as argument, which overlaps `decorated` which then is used to overlap `cls`'s test_name.

Reply:
Removed the first call